Automattic/wp-calypso
# Plans Grid Refactor: Lift action button redirect outside of grid so that `flowName` prop can be removed and cleanup enterprise woo references

Related to: https://github.com/Automattic/wp-calypso/issues/82335

## Proposed Changes
* Moves the redirect actions for special plans out of the plans grid
* The upgradeClickHandler was modified to return the plan selected.
* Removes the flowName prop from the plans grid and across the component hierarchy
* Cleans up the woo express plus plan and it's references since it has been moved out of the plans grid. (https://github.com/Automattic/wp-calypso/pull/78214)

## Testing Instructions



* Go through the plans flows and make sure a click on enterprise plans redirects to the correct URL
* PLans grid should work well in `/plans` and `/start/plans`
* Vevrify the storage addons and plans get added to the shopping cart accurately.


## Pre-merge Checklist
